src/Microsoft.Identity.Client/Azure/Constants.cs

```diff
@@ -0,0 +1,173 @@
+        ///     which is the ID of the tenant where the account resides
+        /// </summary>
+        public const string AzureTenantIdEnvName = "AZURE_TENANT_ID";
+
+        /// <summary>
+        ///     AzureClientIdEnvName is the name of the environment variable which holds a string representation for a GUID
+        ///     ClientId (application ID) of the application
+        /// </summary>
+        public const string AzureClientIdEnvName = "AZURE_CLIENT_ID";
+
+        /// <summary>
+        ///     AzureClientSecretEnvName is the name of the environment variable which holds a string representation of the
+        ///     secret for an application
+        /// </summary>
+        public const string AzureClientSecretEnvName = "AZURE_CLIENT_SECRET";
+
+
+        /// <summary>
+        ///     AzureCertificateEnvName is the name of the environment variable which holds a base64 encoded string
+        ///     representation of an X509 certificate which is the secret for an application
+        /// </summary>
+        public const string AzureCertificateEnvName = "AZURE_CERTIFICATE";
+
+        /// <summary>
+        ///     AzureCertificateThumbprintEnvName is the name of the environment variable which holds a string
+        ///     representation of the thumbprint of a certificate stored in the Windows Certificate Store
+        /// </summary>
+        public const string AzureCertificateThumbprintEnvName = "AZURE_CERTIFICATE_THUMBPRINT";
+
+        /// <summary>
+        ///     AzureCertificateStoreEnvName is the name of the environment variable which holds the name of the certificate
+        ///     store
+        /// </summary>
+        public const string AzureCertificateStoreEnvName = "AZURE_CERTIFICATE_STORE";
+
+        /// <summary>
+        ///     AzureCertificateStoreLocationEnvName is the name of the environment variable which holds the location of the
+        ///     certificate store (CurrentUser or LocalMachine)
+        /// </summary>
+        public const string AzureCertificateStoreLocation = "AZURE_CERTIFICATE_STORE_LOCATION";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on AppService
+        /// </summary>
+        public const string ManagedIdentityAppServiceApiVersion = "2017-08-01";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on Virtual Machines
+        /// </summary>
+        public const string ManagedIdentityVMApiVersion = "2018-02-01";
+
+        /// <summary>
+        /// Managed Identity loopback IP address
+        /// </summary>
+        public const string ManagedIdentityLoopbackAddress = "169.254.169.254";
+
+        /// <summary>
+        /// Managed Identity metadata endpoint
+        /// </summary>
+        public const string ManagedIdentityMetadataEndpoint = "http://" + ManagedIdentityLoopbackAddress + "/metadata";
```

Comment:
I'm going to assume that the answer is no, but possible to use `https` here?

Reply:
Be a lot cooler if you could: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token